### DISCUSSION TITLE
New Home Assistant 2023.7.1 Error "indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)"

### DISCUSSION THREAD

**Discussion by MarcoAVoegele** [spali/go-rscp#40] [Q&A]
Hi Spali,
I just updated the Home Assistant version and now it does no longer work:
"
sensor:
  - platform: command_line
    command: "/config/packages/e3dc/e3dc.sh"
    name: e3dc
    command_timeout: 1
    scan_interval: 1
    json_attributes:
      - result
    value_template: "OK"
  
  - platform: template
    sensors:
      e3dc_grid_power:
        friendly_name: E3DC Netz
        device_class: power
        value_template: "{{ state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] }}"
        unit_of_measurement: W
"
Home Assistant Log:
"
...
ValueError: Sensor sensor.e3dc_grid_power has device class 'power', state class 'None' unit 'W' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)
"
Do you have any suggestions?

Thank you,

Marco

**Comment by spali** [comment]
Try to append a int conversion `int(0)`:
```
value_template: "{{ state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] | int(0) }}"
```
I would recommend this anyway for all templates that expect a number.


I go even so far to control the value if something unexpected returns:
```
value_template: >-
    {{
            (state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] | int(0))
            if state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] is number
            else 'unknown'
    }}
```

**Reply by MarcoAVoegele** [reply]
Thank you... Does not help...
If I call the script directly it returns:
"
➜  e3dc ./e3dc.sh
{"result":{"EMS_POWER_PV":1723,"EMS_POWER_BAT":0,"EMS_POWER_HOME":1191,"EMS_POWER_GRID":-7502,"EMS_POWER_ADD":-6970,"EMS_AUTARKY":99.99996,"EMS_SELF_CONSUMPTION":13.747808,"EMS_BAT_SOC":100,"EMS_EXT_SRC_AVAILABLE":1,"EMS_EMERGENCY_POWER_STATUS":2,"EMS_GET_POWER_SETTINGS":{"EMS_POWER_LIMITS_USED":false,"EMS_MAX_CHARGE_POWER":1500,"EMS_MAX_DISCHARGE_POWER":1500,"EMS_DISCHARGE_START_POWER":95,"
So the values are there....

e3dc.yaml:
sensor:
  - platform: command_line
    command: "/config/packages/e3dc/e3dc.sh"
    name: e3dc
    command_timeout: 1
    scan_interval: 1
    json_attributes:
      - result
    value_template: "OK"
  
  - platform: template
    sensors:
      e3dc_grid_power:
        friendly_name: E3DC Netz
        device_class: power
        value_template: "{{ state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] | int(0) }}"
        unit_of_measurement: W

Log:
"...
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 590, in state
    numerical_value = int(value)
                      ^^^^^^^^^^
ValueError: invalid literal for int() with base 10: ''
...
ValueError: Sensor sensor.e3dc_grid_power has device class 'power', state class 'None' unit 'W' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)

**Reply by spali** [reply]
I'm currently still on 2023.6.3, can't test it currently with the new release.
Try remove the quotes `"` around the template.

**Comment by MarcoAVoegele** [comment]
Thank you, did also not help...
"Cannot quick reload all YAML configurations because the configuration is not valid: Error loading /config/configuration.yaml: while scanning for the next token found character '|' that cannot start any token in "/config/packages/e3dc.yaml", line 16, column 82"

**Reply by spali** [reply]
can you post your full `e3dc.yaml` file? But please formatted within tripple backticks.

**Reply by MarcoAVoegele** [reply]
I needed to pause the project. I rebootet the system and all works with the 
e.g. value_template: "{{ state_attr('sensor.e3dc', 'result')['EMS_POWER_GRID'] | int(0) }}"
Thank you very much for your great support.
